x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -420,109 +420,128 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    public static RoleDescriptor parse(String name, BytesReference source, boolean allow2xFormat, XContentType xContentType)
-        throws IOException {
-        return parse(name, source, allow2xFormat, xContentType, true);
+    public static Parser parser() {
+        return new Parser();
     }
 
-    public static RoleDescriptor parse(
-        String name,
-        BytesReference source,
-        boolean allow2xFormat,
-        XContentType xContentType,
-        boolean allowRestriction
-    ) throws IOException {
-        assert name != null;
-        try (XContentParser parser = createParser(source, xContentType)) {
-            return parse(name, parser, allow2xFormat, allowRestriction);
-        }
-    }
+    public static final class Parser {
```

Comment:
The main change is pulling `parse` method in the `RoleDescriptor.Parser` class and making allow2xFormat and allowRestriction properties of the parser instead of parameters of the method.